specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/associatedTenant.json

```diff
@@ -0,0 +1,377 @@
+            "schema": {
+              "$ref": "./types.json#/definitions/ArmErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "x-ms-examples": {
+          "AssociatedTenantsCreateOrUpdate": {
+            "$ref": "./examples/associatedTenantsCreateOrUpdate.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "AssociatedTenant": {
+      "description": "An associated tenant.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "./types.json#/definitions/ArmResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "The properties of a(n) AssociatedTenant",
+          "allOf": [
+            {
+              "$ref": "#/definitions/AssociatedTenantProperties"
+            }
+          ],
+          "x-ms-client-flatten": true
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "AssociatedTenantListResult": {
+      "description": "A container for a list of resources",
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "The link (url) to the next page of results.",
+          "type": "string",
+          "readOnly": true
+        },
+        "value": {
+          "description": "The list of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AssociatedTenant"
+          },
+          "readOnly": true,
+          "x-ms-identifiers": [
+            "id"
+          ]
+        }
+      }
+    },
+    "AssociatedTenantProperties": {
+      "description": "An associated tenant.",
+      "type": "object",
+      "properties": {
+        "friendlyName": {
+          "description": "The friendly name assigned to the associated tenant.",
+          "type": "string"
+        },
+        "tenantId": {
+          "description": "The ID that uniquely identifies a tenant.",
+          "type": "string"
+        },
+        "billingManagementState": {
+          "description": "The state determines whether users from the associated tenant can be assigned roles for commerce activities like viewing and downloading invoices, managing payments, and making purchases.",
+          "enum": [
+            "Unknown",
+            "NotAllowed",
+            "Active",
+            "Revoked"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "BillingManagementTenantState",
+            "modelAsString": true
+          }
+        },
+        "provisioningState": {
+          "description": "The state determines whether subscriptions and licenses can be provisioned in the associated tenant. It can be set as 'Pending' from null, null from 'Pending', or null from 'Active'",
+          "enum": [
+            "Unknown",
+            "NotRequested",
```

Comment:
Should the property just be nullable? I.e. if someone didn't request provisioning access, then simply don't have a provisioning state value at all.

Reply:
In initial designs and implementations, the state properties were nullable, but it prohibited sort/filter behavior on the enum values.